### PR TITLE
Rosetta/coin metadata cache

### PR DESCRIPTION
**Pull request by nikos-kitmeridis**
## Description 

Adds a Currency cache in order to get information about the coin types when processing transfer transactions.
Also removes wrong balance change info about SUI that is created due to the lack of the coin type when processing the transaction commands in specific cases.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-rosetta/src/types.rs [hunk 1]
```diff
@@ -899,6 +956,24 @@ impl InternalOperation {
                 builder.pay_sui(recipients, amounts)?;
```
Comment (patrickkuo) [inline crates/sui-rosetta/src/types.rs:956]: in order to not misinterpret a transaction as pay sui in rosetta, we might want to change this to include the coin type tags as well, maybe we can have a follow up PR once we have unblocked CB's integration?

lets add a TODO here for now.

File: crates/sui-rosetta/src/lib.rs [hunk 2]
```diff
@@ -99,3 +111,40 @@ impl RosettaOfflineServer {
         axum::serve(listener, app).await.unwrap();
     }
 }
+
+#[derive(Clone)]
+pub struct CoinMetadataCache {
+    client: SuiClient,
+    metadata: Arc<Mutex<HashMap<TypeTag, Currency>>>,
```
Comment (patrickkuo) [inline crates/sui-rosetta/src/lib.rs:118]: this cache will grow if we have a lot of different coin types, I am not sure if this will cause any problem but we might want to use `LruCache` instead to make sure we won't run out of memory.

File: crates/sui-rosetta/src/operations.rs [hunk 3]
```diff
@@ -259,7 +292,7 @@ impl Operations {
                 }
                 SuiArgument::GasCoin => (),
                 // Might not be a SUI coin
-                SuiArgument::Input(_) => return None,
+                SuiArgument::Input(_) => (),
```
Comment (patrickkuo) [inline crates/sui-rosetta/src/operations.rs:295]: why is this change needed?

File: crates/sui-rosetta/src/lib.rs (hunk 2 above)
Comment (nikos-kitmeridis) [reply crates/sui-rosetta/src/lib.rs:118]: Leaving it like this for now to unblock Coinbase integration testing but created a ticket for that which we can take care of soon.
https://mysten.atlassian.net/browse/SEBP-97

File: crates/sui-rosetta/src/operations.rs (hunk 3 above)
Comment (nikos-kitmeridis) [reply crates/sui-rosetta/src/operations.rs:295]: Without this change, in case a pay (coin, other than SUI) operation, this method return early at this line without the required amount which is calculated a few lines below
```
 let value: u64 = match *amount {
                        SuiArgument::Input(i) => {
                            u64::from_str(inputs[i as usize].pure()?.to_json_value().as_str()?)
                                .ok()?
                        }
                        ...
```

File: crates/sui-rosetta/src/types.rs (hunk 1 above)
Comment (nikos-kitmeridis) [reply crates/sui-rosetta/src/types.rs:956]: Not sure what exactly you have in mind here. Do you suggest to replace with `pay` instead? Like it happens for other coins now?

Comment (patrickkuo) [reply crates/sui-rosetta/src/types.rs:956]: we could add a type tag arg to the PTB like other txs, so when rosetta read back the TransactionData it will know it's a pay sui transfer.

Comment (patrickkuo) [reply crates/sui-rosetta/src/types.rs:956]: if there are no type tag arg (transactions created outside rosetta), we can parse it as a move call transaction without mistakenly parse into a pay sui tx
